pkg/v1/remote/image.go

```diff
@@ -42,30 +42,63 @@ type remoteImage struct {
 	config       []byte
 }
 
+type ImageOption func(*ImageOpener) error
+
 var _ partial.CompressedImageCore = (*remoteImage)(nil)
 
-// Image accesses a given image reference over the provided transport, with the provided authentication.
-func Image(ref name.Reference, auth authn.Authenticator, t http.RoundTripper) (v1.Image, error) {
-	scopes := []string{ref.Scope(transport.PullScope)}
-	tr, err := transport.New(ref.Context().Registry, auth, t, scopes)
+type ImageOpener struct {
```

Comment:
nit: I wonder if we could get away with making `ImageOpener` and its methods private (for now).

e.g. `ImageOption` and `WithFoo` would be public, but `ImageOpener` and its `SetFoo` methods would be private (accessed exclusively thru `WithFoo`).

I think this makes the public surface of this package a bit neater.

Reply:
yeah there's no reason why we can't do this